src/components/AddProtocolModal/AddProtocolModal.jsx

```diff
@@ -0,0 +1,250 @@
+                    minRentingDurationUnits: convertedValue,
+                    maxRentingDurationUnits: convertedValue + 1,
+                });
+            }
+        }
+
+        if (name === 'maxRentingDurationUnits') {
+            const convertedValue = Number(value);
+
+            if (convertedValue <= data.minRentingDurationUnits) {
+                return;
+            }
+        }
+
+        setData({
+            ...data,
+            [name]: value.trim(),
+        });
+    };
+
+    const handleSubmit = () => {
+        const { utilityAmount, rentalFeeAmount, collateralAmount, gracePeriodDuration } = data;
+
+        const possibleErrors = {
+            utilityAmount: utilityAmount ? '' : 'Required',
+            rentalFeeAmount: rentalFeeAmount ? '' : 'Required',
+            collateralAmount: collateralAmount ? '' : 'Required',
+            gracePeriodDuration: gracePeriodDuration ? '' : 'Required',
+        };
+
+        if (Object.values(possibleErrors).join('')) {
+            setErrors({ ...errors, ...possibleErrors });
+            return;
+        }
+
+        const processedData = {
+            ...data,
+            rentalFeeAmount: BigInt(+data.rentalFeeAmount),
+            collateralAmount: BigInt(+data.rentalFeeAmount),
+            minRentingDurationUnits: BigInt(+data.rentalFeeAmount),
+            maxRentingDurationUnits: BigInt(+data.rentalFeeAmount),
+        };
+
+        console.log('Success: ', data);
+        // TODO: 2. Clear state after the tx is successfully done
+        // const onModalClose = () => {
+        //     setData(defaultData);
+        //     setErrors(defaultErrors);
+        //     onClose();
+        // };
+        onModalClose();
+        createRentalKeplr(processedData);
+        showSnackbar('Form submitted successfully', 'warning');
+
+        // setSubmittedData(processedData);
+        // RentalCreator(processedData);
+    };
+
+    return (
+        open && (
+            <div className="add-protocol-modal">
+                {submittedData && <RentalCreator data={submittedData} onSubmit={() => setSubmittedData(null)} />}
+                <header>
+                    <h4>Add Crubble Protocol</h4>
+                    <button onClick={onModalClose}>x</button>
+                </header>
+                <section>
+                    <div className="modal-column">
+                        <div>
+                            <p className="label">Choose the NFT you want to rent</p>
+                            <Select
+                                name="utilityAmount"
+                                onChange={handleChange}
+                                value={data.utilityAmount}
+                                native={true}
+                            >
+                                <option value="" disabled>
+                                    Utility Amount
+                                </option>
+                                <option value="ERTP">ERTP</option>
```

Comment:
We have to show the [purse balances](https://github.com/CrabblePitch/crabble-ui/blob/4cf6c9b0f8ff96020128b6f9b29852c0683a066f/src/CreateRentalKeplr.jsx#L15-L17) here.